src/main/java/frc/robot/subsystems/Swerve/SwerveSubsystem.java

```diff
@@ -67,50 +73,88 @@ public SwerveModulePosition[] getPositions() {
         for (int i = 0; i < 4; i++) pos[i] = modules[i].getPosition();
         return pos;
     }
+	public ChassisSpeeds getChassisSpeeds(double xVelocity, double yVelocity, double rotationalVelocity) {
+		return ChassisSpeeds.discretize(ChassisSpeeds.fromFieldRelativeSpeeds(xVelocity, yVelocity, rotationalVelocity, POSE_ESTIMATOR.getEstimatedPose().getRotation()),0.02);
+}
 	public void drive(ChassisSpeeds speeds) {
-		
-		
+	SwerveModuleState[] states = kinematics.toSwerveModuleStates(speeds);
+		for (int i = 0; i < modules.length; i++) {
+		modules[i].setState(states[i]);
+}
+
+	
+}
+
+		/**
+	 * Get the closest angle between the given angles.
+	 */
+	public double closestAngle(double a, double b)
+	{
+			// get direction
+			double dir = (b % 360.0) - (a % 360.0);
+
+			// convert from -360 to 360 to -180 to 180
+			if (Math.abs(dir) > 180.0)
+			{
+					dir = -(Math.signum(dir) * 360.0) + dir;
+			}
+			return dir;
+	}
+	public  Pose2d getCurrentPose(){
+		return POSE_ESTIMATOR.getEstimatedPose();
+	}
+
+	public void drive(double xVelocity, double yVelocity, double rotationalVelocity) {
+		ChassisSpeeds speeds =
+				ChassisSpeeds.fromFieldRelativeSpeeds(xVelocity, yVelocity, rotationalVelocity, POSE_ESTIMATOR.getEstimatedPose().getRotation());
+		speeds = ChassisSpeeds.discretize(speeds, 0.02);
 		SwerveModuleState[] states = kinematics.toSwerveModuleStates(speeds);
 		for (int i = 0; i < modules.length; i++) {
 			modules[i].setState(states[i]);
 		}
 	}
 
+	public Command followPathCommand(String pathName, double xVelocity, double yVelocity, double rotationalVelocity) {
+        PathPlannerPath path = PathPlannerPath.fromPathFile(pathName);
+
+        return new FollowPathHolonomic(
+                path,
+                this::getCurrentPose, // Robot pose supplier
```

Comment:
In here is more clear to just use POSE_ESTIMATOR::getEstimatedPose

Reply:
The pathplannerlib documentation says to only reference the subsystem the method is on for any functions needed.